lib/util/cache/repository/impl/repository-cache-impl.ts

```diff
@@ -1,43 +1,36 @@
 import { promisify } from 'util';
 import zlib from 'zlib';
 import hasha from 'hasha';
-import upath from 'upath';
-import { GlobalConfig } from '../../../../config/global';
+import type { RepositoryCacheType } from '../../../../config/types';
 import { logger } from '../../../../logger';
-import { outputCacheFile, readCacheFile } from '../../../fs';
 import {
   CACHE_REVISION,
   isValidRev10,
   isValidRev11,
   isValidRev12,
 } from '../common';
-import type { RepoCacheRecord } from '../types';
+import type { CacheClient } from '../types';
 import { RepoCacheBase } from './base';
+import { CacheClientFactory } from './client/cache-client-factory';
 
 const compress = promisify(zlib.brotliCompress);
 const decompress = promisify(zlib.brotliDecompress);
 
-export class LocalRepoCache extends RepoCacheBase {
+export class RepositoryCacheImpl extends RepoCacheBase {
+  private cacheClient: CacheClient;
```

Comment:
This cache implementation class now contains only the logic related to handling the cached data. i.e. encoding/decoding.
saving/writing is done by the cache client.